bin/NativeTests/UnicodeTextTests.cpp

```diff
@@ -84,4 +102,10 @@ namespace UnicodeTextTest
         Test(_u("20sTRing"), _u("st2ring"), -1);
         Test(_u("st3rING"), _u("st2riNG"), 1);
     }
+
+    TEST_CASE("LogicalCompareString_EmbeddedNullCharacters", "[UnicodeText]")
+    {
+        TestNullCharacters(_u("\0\0ab\0\0c123def\0\0"), 15, _u("abc123def"), 9, 0);
```

Comment:
I am honestly surprised that these should compare equal -- I just checked firefox and they agree, but I would have thought "\0" would be less than "a" and thus left should be less than right, _or_ left should be greater than right since its longer. Good test case!